### PR TITLE
http: init `NewDefaultPDServiceDiscovery` and refine http client tests

### LINKED ISSUE #7721: “pd-ctl cluster” failed with error "Failed to get the cluster information: [PD:client:ErrClientNoAvailableMember]no available member"
### What did you do?
run "pd-ctl cluster"

### What did you expect to see?
cmd can success

### What did you see instead?
pd-ctl cluster failed with error "Failed to get the cluster information: [PD:client:ErrClientNoAvailableMember]no available member"

### What version of PD are you using (`pd-server -V`)?
./pd-server -V
 Release Version: v8.0.0-alpha
Edition: Community
Git Commit Hash: 3a12148b573944890c6ecc8e270f458a8369bb74
Git Branch: heads/refs/tags/v8.0.0-alpha
UTC Build Time:  2024-01-16 11:36:44
2024-01-17T09:36:16.314+0800

### PR DESCRIPTION
**Pull request by HuSharp**
### What problem does this PR solve?

Issue Number: Close #7721

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Integration test
- Manual test (add detailed scripts or steps below)

### Release note

```release-note
None.
```

### CODE REVIEW

**Review by CabinfeverB — APPROVED** [review]

File: tests/integrations/realcluster/Makefile [hunk 1]
```diff
@@ -48,7 +48,7 @@ kill_cluster:
 		echo $$pid; \
 		kill $$pid; \
 		echo "waiting for cluster to exit..."; \
-		sleep 10; \
+		sleep 30; \
```
[the remark this answers is not in the document]
Comment (HuSharp) [reply tests/integrations/realcluster/Makefile:51]: tiup boostrap three pd, service discovery needs to check all the members, meaning a higher probability of failure than before which it only needed 2379 to work correctly (maybe 10s hadn't exited all pds yet, and the previous ports were still occupied at reboot)

File: client/http/client.go [hunk 2]
```diff
@@ -385,3 +374,20 @@ func NewHTTPClientWithRequestChecker(checker requestChecker) *http.Client {
 		Transport: checker,
 	}
 }
+
+// newClientWithoutServiceDiscovery creates a PD HTTP client with the given PD addresses and TLS config without service discovery.
+func newClientWithoutServiceDiscovery(
```
[the remark this answers is not in the document]
Comment (HuSharp) [reply client/http/client.go:379]: actually, it's used only in tests which is not `init` service discovery to better test in unit test.(I have indicated the code above and did not export it)

**Review by JmPotato — APPROVED** [review]

**Pull request merged**